### PR TITLE
Various Server-Side Apply improvements

### PR DESCRIPTION
**Pull request by apelisse**
@justinsb told me he wasn't sure what permissions were needed to use server-side apply, and we're not doing a great job explaining this today so I added that sub-section about it. I think that's the right location since we also explain that ssa is a patch at that location. I also noticed a caution block that is obsolete since GA. And while I was editing this file, I just normalized every instances to `Server-Side Apply`.

### CODE REVIEW

**Comment by apelisse** [comment]
Ah! you've created a `main` branch, but the `master` branch is neither removed nor an alias to `main`, that's super confusing...

**Comment by apelisse** [comment]
Reference: https://github.com/GoogleContainerTools/kpt/pull/3497#issuecomment-1227228324

**Comment by sftim** [comment]
If any of https://github.com/kubernetes/website/pull/31261 is useful, feel free to borrow from that PR.

**Comment by sftim** [comment]
Also, I don't think this repo has a `master` branch. Check https://github.com/kubernetes/website/tree/master/

**Comment by apelisse** [comment]
Yeah, not sure why I ended-up having a master branch on my remote that doesn't match the main branch.

Also, I love your other PR, but it's fairly big and I'm afraid trying to do too much (merging the two) would significantly delay that quick fix here.

**Comment by sftim** [comment]
What I mean is, you can borrow a paragraph or so if that helps you @apelisse - not a merge of all the work so far.

No problem with you borrowing sentences or paragraphs; I wouldn't object to a `Co-Authored-By:` credit if you do.

**Comment by apelisse** [comment]
Yeah, but I'm afraid that's going to take me a lot of time and possible push-back that I don't really have time to afford right now :-)

**Comment by apelisse** [comment]
OK Fixed conflicts. @sftim, I'll see if I can take some of your PR changes into another smaller PR later if you don't mind.

File: content/en/docs/reference/using-api/server-side-apply.md [hunk 1]
```diff
@@ -465,12 +465,18 @@ kubectl apply --server-side --field-manager=my-manager [--dry-run=server]
 
 ## API Endpoint
 
-With the Server Side Apply feature enabled, the `PATCH` endpoint accepts the
+With the Server-Side Apply feature enabled, the `PATCH` endpoint accepts the
 additional `application/apply-patch+yaml` content type. Users of Server Side
```
Comment (chrisnegus) [inline content/en/docs/reference/using-api/server-side-apply.md:469]:
```suggestion
additional `application/apply-patch+yaml` content type. Users of Server-Side
```

**Review by chrisnegus — CHANGES_REQUESTED** [review]
Found one "Server-Side" you missed. Otherwise lgtm.

File: content/en/docs/reference/using-api/server-side-apply.md (hunk 1 above)
Comment (apelisse) [reply content/en/docs/reference/using-api/server-side-apply.md:469]: nice catch, my very simple search-and-replace didn't catch this case ;-)

Comment (apelisse) [reply content/en/docs/reference/using-api/server-side-apply.md:469]: Actually I found another instance!

Comment (chrisnegus) [reply content/en/docs/reference/using-api/server-side-apply.md]: Nice. That is one reason I like to start and end each sentence on its own line. One's searches are not foiled by the end of a line.

File: content/en/docs/reference/using-api/server-side-apply.md [hunk 2]
```diff
@@ -465,12 +465,18 @@ kubectl apply --server-side --field-manager=my-manager [--dry-run=server]
...
 
-With the Server Side Apply feature enabled, the `PATCH` endpoint accepts the
-additional `application/apply-patch+yaml` content type. Users of Server Side
+With the Server-Side Apply feature enabled, the `PATCH` endpoint accepts the
+additional `application/apply-patch+yaml` content type. Users of Server-Side
 Apply can send partially specified objects as YAML to this endpoint.  When
 applying a configuration, one should always include all the fields that they
 have an opinion about.
 
+### RBAC And permissions
```
Comment (Sea-n) [inline content/en/docs/reference/using-api/server-side-apply.md:474]:
```suggestion
### RBAC and Permissions
```
or
```suggestion
### RBAC and permissions
```

**Review by Sea-n — COMMENTED** [review]
Seems reasonable.

In addition, I think the first two commits (remove & add section) should be squash into one. How do you think?

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-36293--kubernetes-io-main-staging.netlify.app/docs/reference/using-api/server-side-apply/

**Comment by kbhawkey** [comment]
@apelisse , Can you rebase into a single commit?
/lgtm

File: content/en/docs/reference/using-api/server-side-apply.md (hunk 2 above)
Comment (kbhawkey) [reply content/en/docs/reference/using-api/server-side-apply.md:474]: @apelisse , What do you think about the second suggestion?

Comment (apelisse) [reply content/en/docs/reference/using-api/server-side-apply.md:474]: Thanks for the ping, missed that somehow, sending a fix with the second suggestion!

**Comment by apelisse** [comment]
Updated

**Pull request merged**